assembly/src/v1/tokens/stream.rs

```diff
@@ -20,7 +20,23 @@ impl<'a> TokenStream<'a> {
         if source.is_empty() {
             return Err(AssemblyError::empty_source());
         }
-        let tokens = source.split_whitespace().collect::<Vec<_>>();
+
+        let mut in_comment = false;
+        let tokens = source
+            .split_whitespace()
+            .filter(|&token| {
+                if token == "#" {
+                    in_comment = !in_comment;
+                } else if !in_comment {
+                    return true;
+                }
+                return false;
+            })
+            .collect::<Vec<_>>();
```

Comment:
This works! Originally I thought we'd do this by modifying `advance()` method to skip over comments there. This would avoid an extra pass. But this is good enough, and if performance ever becomes an issue, we can always revisit this.

Reply:
Yes, I had originally been planning to do it as the contents were being consumed for that reason, but that requires some other updates to account for the checks on 'begin' and 'end', so I think it doesn't keep it all as neatly within the TokenStream (or requires some lookahead stuff) - that's why I asked previously to confirm where comments are allowed. (I think I misinterpreted your response about doing it in the iterator in TokenStream, though.)

I'd be happy to go through and do it inside advance! I'll push the other updates first though

Reply:
Awesome! I think what we have now is good. Moving logic into `advance()` would require this method to return an error for un-terminated comments, and that would lead to changes in many other places. So, it's a lot of code changes for little benefit.